src/native/fs/fs_napi.cpp

```diff
@@ -1939,6 +1943,52 @@ struct DirOpen : public FSWorker
     }
 };
 
+struct TellDir : public FSWrapWorker<DirWrap>
+{
+    std::string _dir_path;
+    long int _tell_res;
+    TellDir(const Napi::CallbackInfo& info)
+        : FSWrapWorker<DirWrap>(info)
+    {
+        Begin(XSTR() << "TellDir " << DVAL(_wrap->_path));
+    }
+    virtual void Work()
+    {
+        DIR* dir = _wrap->_dir;
+        std::string _dir_path = _wrap->_path;
+        DBG1("FS::Telldir::Work: " << DVAL(_dir_path));
+        _tell_res = telldir(dir);
+        if (_tell_res == -1) SetSyscallError();
+    }
+    virtual void OnOK()
+    {
+        DBG0("FS::Telldir::OnOK: " << DVAL(_dir_path) << DVAL(_tell_res));
+        Napi::Env env = Env();
+        auto res = Napi::Object::New(env);
+        res["telldir_res"] = Napi::Number::New(env, _tell_res);
```

Comment:
Do we want to return a number or an object?
if a number, the return value should look like - 
```
Napi::Env env = Env();
auto res = Napi::Number::New(env, _tell_res);
_deferred.Resolve(res);
```

Reply:
@romayalon @naveenpaul1 @eshelmarc 

First I want to verify if we might need to support 64bit dir cookie - I see some traces of such implementations, but no clear documentation so might not be posix:
- [IBM AIX-telldir64](https://www.ibm.com/docs/en/aix/7.3?topic=o-opendir-readdir-telldir-seekdir-rewinddir-closedir-opendir64-readdir64-telldir64-seekdir64-rewinddir64-closedir64-fdopendir-subroutine)
- [Ext4 64 bit hash breaks 32 bit glibc 2.28+](https://bugzilla.kernel.org/show_bug.cgi?id=205957) 

However libc [telldir](https://sourceware.org/git/?p=glibc.git;a=blob;f=dirent/dirent.h;h=b33d6296e3b1889414a604eb9e99f96ee60c04b6;hb=HEAD#l220)/[seekdir](https://sourceware.org/git/?p=glibc.git;a=blob;f=dirent/dirent.h;h=b33d6296e3b1889414a604eb9e99f96ee60c04b6;hb=HEAD#l217) has fixed this type on 32bit long int.

In any case, I think our interface should return this as an opaque instead of a js number - I would go with a hex string of the bytes that compose this cookie number, and parse back on seekdir. We can also return a Buffer, but I don't know if this is really better - js is notoriously good with managing short strings.

Reply:
OK, not sure why but I got confused here and was corrected - long int is int64 on a 64bit system.
So this is why libc used it, and this is also the reason why we can't just stick it in js number (which is a float and represents ints with up to 53 bits).

So I suggest is to change this to return js BigInt which can hold 64bit int without a problem.
See https://github.com/nodejs/node-addon-api/blob/main/doc/bigint.md